legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalLockProviderAbstract.java

```diff
@@ -0,0 +1,128 @@
+import org.finos.legend.engine.persistence.components.util.LockInfoDataset;
+import org.finos.legend.engine.persistence.components.util.LockInfoUtils;
+import org.finos.legend.engine.persistence.components.util.PlaceholderValue;
+import org.finos.legend.engine.persistence.components.util.SqlLogging;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Default;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.util.*;
+
+import static org.finos.legend.engine.persistence.components.relational.api.RelationalIngestorAbstract.BATCH_START_TS_PATTERN;
+import static org.finos.legend.engine.persistence.components.transformer.Transformer.TransformOptionsAbstract.DATE_TIME_FORMATTER;
+
+@Value.Immutable
+@Value.Style(
+        typeAbstract = "*Abstract",
+        typeImmutable = "*",
+        jdkOnly = true,
+        optionalAcceptNullable = true,
+        strictBuilder = true
+)
+public abstract class RelationalLockProviderAbstract
+{
+    //---------- FIELDS ----------
+    public abstract RelationalSink relationalSink();
+
+    @Default
+    public CaseConversion caseConversion()
+    {
+        return CaseConversion.NONE;
+    }
+
+    @Default
+    public SqlLogging sqlLogging()
+    {
+        return SqlLogging.DISABLED;
+    }
+
+    @Default
+    public Clock executionTimestampClock()
+    {
+        return Clock.systemUTC();
+    }
+
+    @Value.Derived
+    protected Transformer<SqlGen, SqlPlan> transformer()
+    {
+        TransformOptions.Builder builder = TransformOptions.builder().executionTimestampClock(executionTimestampClock());
+        relationalSink().optimizerForCaseConversion(caseConversion()).ifPresent(builder::addOptimizers);
+        TransformOptions transformOptions = builder.build();
+        return new RelationalTransformer(relationalSink(), transformOptions);
+    }
+
+    // ---------- Private Fields ----------
+    private static final Logger LOGGER = LoggerFactory.getLogger(RelationalLockProvider.class);
+
+    // ------API-----
+    public void createAndInitialize(Executor<SqlGen, TabularData, SqlPlan> executor, LockInfoDataset lockDataset, String tableName)
+    {
+        executor.setSqlLogging(sqlLogging());
+        LOGGER.info("Creating and initializing Lock dataset: " + lockDataset.name());
+        Map<String, PlaceholderValue> placeHolderKeyValues = new HashMap<>();
+        placeHolderKeyValues.put(BATCH_START_TS_PATTERN, PlaceholderValue.of(LocalDateTime.now(executionTimestampClock()).format(DATE_TIME_FORMATTER), false));
+        executor.executePhysicalPlan(createAndInitializeLockSqlPlan(lockDataset, tableName), placeHolderKeyValues);
+    }
+
+    public void lock(Executor<SqlGen, TabularData, SqlPlan> executor, LockInfoDataset lockDataset)
+    {
+        executor.setSqlLogging(sqlLogging());
+        LOGGER.info("Acquiring lock using LockInfoDataset: " + lockDataset.name());
```

Comment:
Same as above - Logging happening before case conversion, so we might see a mismatch between the logged name and the actual name?

Reply:
I think it's fine, will be expensive to convert this just for logging. It represents the value that user passed.
Will print the caseConversion param here.